packages/server-wallet/src/protocols/state.ts

```diff
@@ -0,0 +1,46 @@
+import {SignedStateWithHash} from '@statechannels/wallet-core';
+
+import {Either, Option} from '../fp';
+
+import {ProtocolAction} from './actions';
+
+/*
+The ChannelData type is the data that protocols need about the channel to decide what to do next.
+*/
+export type ChannelData = {
+  channelId: string;
+  myIndex: 0 | 1;
+  supported?: SignedStateWithHash;
+  latest?: SignedStateWithHash;
+  latestSignedByMe?: SignedStateWithHash;
+};
+
+export type Stage = 'Missing' | 'PrefundSetup' | 'PostfundSetup' | 'Running' | 'Final';
```

Comment:
"Closing" might be a stage?

Reply:
Also, "challenging" or "responding", unless it's  included within Running

Reply:
>"Closing" might be a stage?

A channel might be closing, but a state is either final or not final.

Likewise,
> Also, "challenging" or "responding", unless it's included within Running

A channel might be "challenging" or "responding", but that is not a property of a state. It is a function of the stage of `latest`, `latestSignedByMe`, and the adjudicator state (which is not yet recorded).